lumicks/pylake/kymotracker/kymotrack.py

```diff
@@ -486,10 +507,15 @@ def __len__(self):
         return len(self._src)
 
     def extend(self, other):
+        other = [other] if isinstance(other, KymoTrack) else other
+
+        if len(self):
```

Comment:
It's possible to end up with an empty group (if you have really bad tracking parameters for instance). Without testing for length this will throw an `IndexError` because the empty group tries to access `self[0]`. However, I think we should still allow an empty group to be extended

If you unintentionally try to access `group._kymo` or `group._channel` you do get the original `IndexError` thrown from `__getitem__()`. I thought about adding a more descriptive error message in these properties themselves but I think for internal API it's a bit overkill?

Reply:
Ah right. I meant more why are you checking the length specifically. All we want to check is empty, so can't we just check `if self._src:` since an empty list is falsy?